exp/controllers/rosamachinepool_controller.go

```diff
@@ -123,7 +123,21 @@ func (r *ROSAMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	controlPlane := &rosacontrolplanev1.ROSAControlPlane{}
 	if err := r.Client.Get(ctx, controlPlaneKey, controlPlane); err != nil {
 		log.Info("Failed to retrieve ControlPlane from MachinePool")
-		return reconcile.Result{}, nil
+
+		if apierrors.IsNotFound(err) && rosaMachinePool.DeletionTimestamp != nil && !rosaMachinePool.DeletionTimestamp.IsZero() {
+			// Remove the finalizer as the controlPlan is deleted and deleting rosaMachinePool.
+			patcher, err := scope.NewRosaMachinePoolPatchHelper(rosaMachinePool, r.Client)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			if controllerutil.RemoveFinalizer(rosaMachinePool, expinfrav1.RosaMachinePoolFinalizer) {
+				if err = patcher.Patch(context.TODO(), rosaMachinePool); err != nil {
+					return reconcile.Result{}, err
+				}
+			}
+			return ctrl.Result{}, nil
+		}
+		return reconcile.Result{}, err
```

Comment:
```suggestion
		return ctrl.Result{}, err
```
let's keep it consistent and always use `ctrl.Result{}`